node/src/reactor/validator.rs

```diff
@@ -367,6 +371,24 @@ impl reactor::Reactor for Reactor {
             init_consensus_effects,
         ));
 
+        // set timeout to 5 minutes after now, or 5 minutes after genesis, whichever is later
+        let five_minutes = TimeDiff::from_str("5minutes").unwrap();
+        let later_timestamp = cmp::max(
+            Timestamp::now(),
+            chainspec_loader
+                .chainspec()
+                .genesis
+                .highway_config
+                .genesis_era_start_timestamp,
+        );
+        let timer_duration = later_timestamp + five_minutes - Timestamp::now();
```

Comment:
I'd call `now()` only once — not sure if it's guaranteed that this is monotonous?

Reply:
A good question - probably not, since it's based on the system clock.